kpatch-build/create-diff-object.c

```diff
@@ -3676,6 +3681,206 @@ static void kpatch_create_callbacks_objname_rela(struct kpatch_elf *kelf, char *
 	}
 }
 
+static void kpatch_set_pfe_link(struct kpatch_elf *kelf)
+{
+	struct section* sec;
+	struct rela *rela;
+
+	if (!kelf->has_pfe)
+		return;
+
+	list_for_each_entry(sec, &kelf->sections, list) {
+		if (strcmp(sec->name, "__patchable_function_entries")) {
+			continue;
+		}
+
+		if (!sec->rela) {
+			continue;
+		}
+		list_for_each_entry(rela, &sec->rela->relas, list) {
+			rela->sym->sec->pfe = sec;
+		}
+	}
+}
+
+static void kpatch_create_pfe_sections(struct kpatch_elf *kelf)
+{
+	int nr, index;
+	struct section *sec, *relasec;
+	struct symbol *sym;
+	struct rela *pfe_rela;
+
+	nr = 0;
+	list_for_each_entry(sym, &kelf->symbols, list)
+		if (sym->type == STT_FUNC && sym->status != SAME &&
+		    sym->has_func_profiling)
+			nr++;
+
+	/*
+	 * We will create separate __patchable_function_entries
+	 * sections for each symbols.
+	 */
+	kelf->has_pfe = true;
+	sec = find_section_by_name(&kelf->sections, "__patchable_function_entries");
```

Comment:
This has no purpose apparently?

Also since this function is very similar to kpatch_create_mcount_sections() it would be a good idea to mirror that logic as much as possible so it's easier to review.

Reply:
Oops, looks like `sec` and `relasec` assignments are just overwritten here.

With respect to kpatch_create_mcount_sections() - yes, ideally there will be a single function, but I found it much easier to debug and review a standalone copy, at least to start with.  Once the pfe functionality is set (and the commonality determined), I can merge the two functions back together.

Reply:
They don't necessarily need to be combined into a single function, if it's easier to keep them apart.  I was just looking for more symmetry between the two functions so it's easier to see they're correct.